contrib/cli/great_expectations_contrib/package.py

```diff
@@ -110,22 +113,97 @@ def update_package_state(self) -> None:
         """
         Parses diagnostic reports from package Expectations and uses them to update JSON state
         """
-        diagnostics = self._retrieve_package_expectations_diagnostics()
+        diagnostics = (
+            GreatExpectationsContribPackageManifest.retrieve_package_expectations_diagnostics()
+        )
         self._update_attrs_with_diagnostics(diagnostics)
 
-    def _update_attrs_with_diagnostics(self, diagnostics: List[Diagnostics]) -> None:
-        # TODO: Write logic to assign values to attrs
-        # This is a black box for now
-        # for diagnostic in diagnostics:
-        #     pass
-        raise NotImplementedError
-
-    def _retrieve_package_expectations_diagnostics(self) -> List[Diagnostics]:
+    def _update_attrs_with_diagnostics(
+        self, diagnostics: List[ExpectationDiagnostics]
+    ) -> None:
+        self._update_expectations(diagnostics)
+        self._update_dependencies("requirements.txt")
+        self._update_contributors(diagnostics)
+
+    def _update_expectations(self, diagnostics: List[ExpectationDiagnostics]) -> None:
+        expectations = []
+        status = {maturity.name: 0 for maturity in Maturity}
+
+        for diagnostic in diagnostics:
+            expectation = RenderedExpectation(
+                name=diagnostic.description.snake_name,
+                tags=diagnostic.library_metadata.tags,
+                supported=[],
+                status=diagnostic.maturity_checklist,  # Should be converted to the proper type
```

Comment:
@abegong it seems as though `ExpectationDiagnosticMaturityMessages` and `ExpectationCompletenessChecklist` have quite a bit of overlap. Can we leverage the existing type?